Sources/SourceKit/Workspace.swift

```diff
@@ -27,19 +27,32 @@ import SKSwiftPMWorkspace
 /// Typically a workspace is contained in a root directory.
 public final class Workspace {
 
-  /// The root directory of the workspace.
-  public let rootPath: AbsolutePath?
+  /// Workspace configuration per root path
+  public struct Configuration {
 
-  public let clientCapabilities: ClientCapabilities
+    /// The root directory of the workspace.
+    public var rootPath: AbsolutePath? = nil
+
+    /// The build settings provider to use for documents in this workspace.
+    public var buildSettings: BuildSystem
+
+    /// The index to use for documents in this worspace.
+    public var index: IndexStoreDB? = nil
 
-  /// The build settings provider to use for documents in this workspace.
-  public let buildSettings: BuildSystem
+    /// Build setup
+    public let buildSetup: BuildSetup
+
+    init(rootPath: AbsolutePath? = nil, buildSettings: BuildSystem, index: IndexStoreDB? = nil, buildSetup: BuildSetup = .default) {
+      self.rootPath = rootPath
+      self.buildSettings = buildSettings
+      self.index = index
+      self.buildSetup = buildSetup
+    }
+  }
 
-  /// Build setup
```

Comment:
This patch changed the server to contain multiple workspaces, but this particular change to factor the fields into a Configuration struct seems to assume the opposite: a single workspace with multiple folders.  Can you clarify the intent?

Reply:
Looks like post https://github.com/apple/sourcekit-lsp/pull/65 merge issue.
`Configuration` only encapsulate config settings, doesn't change the previous meaning, does it? The `BuildSetup` though is overriding the setup for all workspaces that is not intended here.

Reply:
> The BuildSetup though is overriding the setup for all workspaces that is not intended here.

After revisiting the code, it looks ok to me (I did refactor it to make the initializer intent clearer: d11029c8a823573cb09f682255d4a7c15fb31b7e).
The `BuildSetup` doesn't matter for ".noRoot" Workspace. Each property is setup per Workspace instance.